src/test/java/io/aiven/kafka/connect/gcs/config/GcsSinkConfigTest.java

```diff
@@ -379,25 +390,89 @@ void gcsCredentialsJson() throws IOException {
         assertEquals("test-client-secret", credentials.getClientSecret());
     }
 
+    /**
+     * This test validates that the NoCredentials are used when default is specified as false. This behaviour
+     * mimics that of the Tiered Storage Manager.
+     */
     @Test
-    void gcsCredentialsExclusivity() throws IOException {
-        final Map<String, String> properties = new HashMap<>();
-        properties.put("gcs.bucket.name", "test-bucket");
+    void gcsCredentialsNoCredentialsWhenDefaultCredentialsFalse() {
+        final Map<String, String> properties = Map.of(
+            GCS_BUCKET_NAME_CONFIG, "test-bucket",
+            GCS_CREDENTIALS_DEFAULT_CONFIG, String.valueOf(false)
+        );
 
-        final URL credentialsResource = Thread.currentThread()
-                .getContextClassLoader()
-                .getResource("test_gcs_credentials.json");
-        final String credentialsJson = Resources.toString(credentialsResource, StandardCharsets.UTF_8);
-        properties.put("gcs.credentials.json", credentialsJson);
-        properties.put("gcs.credentials.path", credentialsResource.getPath());
+        assertConfigDefValidationPasses(properties);
+
+        final GcsSinkConfig config = new GcsSinkConfig(properties);
+
+        final Credentials credentials = config.getCredentials();
+        assertEquals(NoCredentials.getInstance(), credentials);
+    }
+
+    /** Verifies that NoCredentials are used when no credential configurations is supplied. */
+    @Test
+    void gcsCredentialsNoCredentialsWhenNoCredentialsSupplied() {
+        final Map<String, String> properties = Map.of(GCS_BUCKET_NAME_CONFIG, "test-bucket");
+
+        assertConfigDefValidationPasses(properties);
+
+        final GcsSinkConfig config = new GcsSinkConfig(properties);
+
+        final Credentials credentials = config.getCredentials();
+        assertEquals(NoCredentials.getInstance(), credentials);
+    }
+
+    @Test
+    void gcsCredentialsDefault() {
+        final Map<String, String> properties = Map.of(
+            GCS_BUCKET_NAME_CONFIG, "test-bucket",
+            GCS_CREDENTIALS_DEFAULT_CONFIG, String.valueOf(true)
+        );
+
+        assertConfigDefValidationPasses(properties);
+
+        try (MockedStatic<GoogleCredentialsBuilder> mocked = mockStatic(GoogleCredentialsBuilder.class)) {
```

Comment:
do we actually need a mock here? can't we use `GoogleCredentials.getApplicationDefault()` as expected value?

Reply:
Good idea. I already got rid of one mockstatic, let's get rid of this one too.

Reply:
Done